### PR TITLE
Add `show_api` parameter to events, and fix `gr.load()`. Also makes some minor improvements to the "view API" page when running on Spaces

### PR DESCRIPTION
**Pull request by abidlabs**
This PR does several things

1. It adds a `show_api` parameter to each event trigger. This `show_api` parameter controls whether or not the API route is visible in the API docs. Specifically, we set to `False` for internal routes in `gr.Interface()` and `gr.ChatInterface()` instead of setting `api_name=False`, which was causing issues using `gr.load()` with `gr.Interface()` and `gr.ChatInterface()`.

Fixes: #6744
Fixes: #6476 

2. Disallows the `api_name=""`, which was causing issues. If `api_name=""` is passed in, it is treated the same as `api_name=None`, and the function name is used instead. 

Fixes: #4894 

3. Also makes some minor improvements to the View API docs pages. For example, "Named endpoints" is replaced with "endpoints", endpoints are counted correctly, and the whitespace is fixed. 

------

**Testing instructions:**

I've installed this PR in: https://huggingface.co/spaces/gradio/test-gr-load

You can confirm this works by `gr.load()`-in the Space as follows and making sure the loaded Space works fully:

```py
import gradio as gr

io = gr.load("spaces/gradio/test-gr-load")
io.launch()
```

### CODE REVIEW

File: client/python/gradio_client/utils.py [hunk 1]
```diff
@@ -856,9 +856,9 @@ def _json_schema_to_python_type(schema: Any, defs) -> str:
     elif type_ == "null":
         return "None"
     elif type_ == "const":
-        return f"Litetal[{schema['const']}]"
+        return f"Literal[{schema['const']}]"
     elif type_ == "enum":
-        return f"Literal[{', '.join([str(v) for v in schema['enum']])}]"
+        return "Literal[" + ", ".join(["'" + str(v) + "'" for v in schema['enum']]) + "]"
```
Comment (abidlabs) [inline client/python/gradio_client/utils.py:861]: Unrelated fix that just adds quotation marks around literals. So instead of the type hint in the API docs being: `Literal[a]`, it will be correctly displayed as `Literal["a"]`

File: gradio/blocks.py [hunk 2]
```diff
@@ -2252,9 +2255,11 @@ def get_api_info(self):
         """
         config = self.config
         api_info = {"named_endpoints": {}, "unnamed_endpoints": {}}
-        mode = config.get("mode", None)
 
-        for d, dependency in enumerate(config["dependencies"]):
+        for dependency in config["dependencies"]:
```
Comment (abidlabs) [inline gradio/blocks.py:2259]: Simplified this block of code a bit

**Comment by abidlabs** [comment]
Need to add / fix tests, which I'll do tomorrow, but otherwise should be good for review

**Review by hysts — APPROVED** [review]
Thank you! So now `gr.Examples` no longer shows `/load_example` too. Awesome!
LGTM!

**Comment by abidlabs** [comment]
> Thank you! So now gr.Examples no longer shows /load_example too. Awesome!

Yeah that was bugging me as well

**Comment by abidlabs** [comment]
Noticed an issue while adding the tests, will fix

File: js/app/src/api_docs/ApiDocs.svelte [hunk 3]
```diff
@@ -79,9 +79,15 @@
...
-	get_info().then((data) => (info = data));
+	get_info().then((data) => {
+		info = data;
+		console.log("info", info);
+	});
 
-	get_js_info().then((js_api_info) => (js_info = js_api_info));
+	get_js_info().then((js_api_info) => {
+		js_info = js_api_info;
+		console.log("js_info", js_info);
```
[the remark this answers is not in the document]
Comment (abidlabs) [reply js/app/src/api_docs/ApiDocs.svelte:89]: Ah thanks I'll clean that up

File: js/app/src/api_docs/ApiDocs.svelte [hunk 4]
```diff
@@ -182,11 +187,11 @@
 				<InstallSnippet {current_language} />
 
 				{#if Object.keys(info.named_endpoints).length}
-					<h2 class="header">Named Endpoints</h2>
+					<h2 class="header">Endpoints</h2>
```
Comment (freddyaboulton) [inline js/app/src/api_docs/ApiDocs.svelte:190]: The deployed space says "undefined" right underneath the Enpoints header

image

File: js/app/src/api_docs/ApiDocs.svelte [hunk 5]
```diff
@@ -144,13 +148,14 @@
 </script>
 
 {#if info}
-	{#if Object.keys(info.named_endpoints).length + Object.keys(info.unnamed_endpoints).length}
+	{#if dependencies.filter((dependency) => dependency.api_name && dependency.show_api).length}
```
Comment (freddyaboulton) [inline js/app/src/api_docs/ApiDocs.svelte:151]: I think we're writing `dependency.api_name && dependency.show_api` enough times in this file that we can write a helper function in the script tag to improve readability

File: js/app/src/api_docs/ApiDocs.svelte [hunk 6]
```diff
@@ -79,9 +79,13 @@
 
 	let js_info: Record<string, any>;
 
-	get_info().then((data) => (info = data));
+	get_info().then((data) => {
```
Comment (freddyaboulton) [inline js/app/src/api_docs/ApiDocs.svelte:82]: Unrelated but I think it'd be nice if we fixed the url for deployed spaces. I think there's an issue open

image

File: client/python/gradio_client/utils.py [hunk 7]
```diff
@@ -856,9 +856,11 @@ def _json_schema_to_python_type(schema: Any, defs) -> str:
     elif type_ == "null":
         return "None"
     elif type_ == "const":
-        return f"Litetal[{schema['const']}]"
```
Comment (freddyaboulton) [inline client/python/gradio_client/utils.py:859]: 🤦‍♂️

File: gradio/blocks.py [hunk 8]
```diff
@@ -2252,9 +2276,15 @@ def get_api_info(self):
...
+            if (
+                not dependency["backend_fn"]
```
Comment (freddyaboulton) [inline gradio/blocks.py:2282]: I think this change does not account for tabbed interfaces any longer and so the tabbed interface api docs are broken

image


I ran the `stt_or_tts` demo

**Review by freddyaboulton — COMMENTED** [review]
Nice @abidlabs ! Left some nits. Can be merged once they're addressed - no need to request re-review imo.

File: js/app/src/api_docs/ApiDocs.svelte (hunk 4 above)
Comment (abidlabs) [reply js/app/src/api_docs/ApiDocs.svelte:190]: I actually don't see the footer at all in the deployed Space? How are you seeing this?

image

File: js/app/src/api_docs/ApiDocs.svelte (hunk 5 above)
Comment (abidlabs) [reply js/app/src/api_docs/ApiDocs.svelte:151]: Let me refactor this

File: js/app/src/api_docs/ApiDocs.svelte (hunk 6 above)
Comment (abidlabs) [reply js/app/src/api_docs/ApiDocs.svelte:82]: Yeah I was thinking about fixing this, will do so in a follow-up PR

File: gradio/blocks.py (hunk 8 above)
Comment (abidlabs) [reply gradio/blocks.py:2282]: Interesting, nice catch will take a look

Comment (abidlabs) [reply gradio/blocks.py:2286]: I can't repro this, perhaps it was fixed in my last change -- would you be able to double check if its still an issue for you @freddyaboulton?

File: js/app/src/api_docs/ApiDocs.svelte (hunk 4 above)
Comment (abidlabs) [reply js/app/src/api_docs/ApiDocs.svelte:189]: If you meant the test Space, I just installed the latest version of this branch and its working for me: 
image

**Review by freddyaboulton — APPROVED** [review]
Those two issues I noticed were my bad. LGTM!

**Comment by abidlabs** [comment]
Thank you for the detailed review @freddyaboulton!
